petastorm/reader.py

```diff
@@ -36,29 +34,99 @@
 
 logger = logging.getLogger(__name__)
 
-# For backward compatibility. Would clean this up with package reorganization in the future
-# https://github.com/uber/petastorm/issues/134
-ShuffleOptions = petastorm.shuffle_options.ShuffleOptions
-
 # Make it easier to import as "from petastorm.reader import ReaderV2"
 ReaderV2 = ReaderV2
 
 
+def make_reader(dataset_url, schema_fields=None, reader_pool=None, shuffle_row_groups=True,
+                shuffle_row_drop_partitions=1, predicate=None, rowgroup_selector=None, num_epochs=1,
+                training_partition=None, num_training_partitions=None, cache=None,
```

Comment:
While we are at it, do you want to rename:
num_training_partitions -> shards_count
training_partition -> shard_id
(or some other twist on sharding). Many scenarios folks are using it not only for distributed training. Today, a new case: sharding between spark executors for evaluation.